internal/checkers/regexp.go

```diff
@@ -0,0 +1,42 @@
+package checkers
+
+import (
+	"go/ast"
+
+	"golang.org/x/tools/go/analysis"
+)
+
+// Regexp detects situations like
+//
+//	assert.Regexp(t, regexp.MustCompile(`\[.*\] DEBUG \(.*TestNew.*\): message`), out)
+//	assert.NotRegexp(t, regexp.MustCompile(`\[.*\] TRACE message`), out)
+//
+// and requires
+//
+//	assert.Regexp(t, `\[.*\] DEBUG \(.*TestNew.*\): message`, out)
+//	assert.NotRegexp(t, `\[.*\] TRACE message`, out)
+type Regexp struct{}
+
+// NewRegexp constructs Regexp checker.
+func NewRegexp() Regexp     { return Regexp{} }
+func (Regexp) Name() string { return "regexp" }
+
+func (checker Regexp) Check(pass *analysis.Pass, call *CallMeta) *analysis.Diagnostic {
+	if n := call.Fn.NameFTrimmed; n != "Regexp" && n != "NotRegexp" {
```

Comment:
When checking the same variable multiple times, I would have used a switch here.
I think it would improve readability, especially because these are negation.